### PR TITLE
Implement universal notification mechanism (NEW)

### PR DESCRIPTION
**Pull request by atighineanu**
Without this patch, kured can only notify via slack.

 This patch gives the possibility to send notifications
 across different technologies. Also, this patch makes
 slack-hook-url, slack-username and slack-channel
 deprecated (informed by a warning).
 Also, updated the documentation (Readme).

### CODE REVIEW

**Comment by atighineanu** [comment]
Added to a kind ds:
```
          command:
            - /usr/bin/kured
            - --period=15s
            - --notify-url=rocketchat://<my_local_rc_instance>:3000/<tokenA>/<tokenB>/general

```

this was the result after triggering a reboot on the first kind worker:
```
kured-bot
Bot
12:17 PM
Draining node kind-worker


kured-bot
Bot
12:17 PM
Rebooting node kind-worker
```

**Comment by jackfrancis** [comment]
I can volunteer to do some manual testing of this next week. Ping me @atighineanu if you don't hear back.

**Comment by jackfrancis** [comment]
@atighineanu can you add more detail about the fact that this PR proposes to replace the slack-specific notification implementation with a generic notification interface built upon the `shoutrrr` project?

It would help to provide some detail on the state of that project, including any supporting evidence that it is healthy and committed to sticking around for the long haul.

Thanks!

**Comment by atighineanu** [comment]
> @atighineanu can you add more detail about the fact that this PR proposes to replace the slack-specific notification implementation with a generic notification interface built upon the `shoutrrr` project?
> 
> It would help to provide some detail on the state of that project, including any supporting evidence that it is healthy and committed to sticking around for the long haul.
> 
> Thanks!

I am keeping an eye and I am among the maintainers over there. The team is chill and last time @dholbach found some issue for one of the messaging technologies - I fixed it and it was merged on the same day. 

Also, this specific PR **IS NOT** replacing the slack-specific notification method. First, I thought it would make sense to make a warning about deprecation. IF both flags are used - then notify-url has priority. But one **can** still use the slack-url flag. 
Next step would be to actually replace the slack-url flag in the flagcheck function (and remove the obsolete slack package).

**Comment by dholbach** [comment]
I rebased on `master` and pushed to `docker.io/weaveworks/kured:atighineanu-master-4d715b8`. Will test it in our Dev env.

File: README.md
Comment (dholbach) [inline README.md:220]: Do we want to delete this?

File: README.md
Comment (dholbach) [inline README.md:215]:
```suggestion
### Notifications
```

File: go.mod
Comment (dholbach) [inline go.mod:6]: Is this indented correctly?

Comment (jackfrancis) [reply go.mod:6]: for consistency we want this new entry to be indented by one tab (not four spaces)

**Comment by dholbach** [comment]
`docker.io/weaveworks/kured:atighineanu-master-e95953b` is up for testing.

**Comment by dholbach** [comment]
Screenshot from 2021-03-19 15-11-13

^ This was with `4d715b8`.

File: README.md
Comment (dholbach) [inline README.md:228]: Why is this section being deleted?

File: cmd/kured/main.go
Comment (dholbach) [inline cmd/kured/main.go:103]: I guess we will need to expose this in the helm chart as well, right?

**Comment by dholbach** [comment]
Screenshot from 2021-03-19 15-20-37

In Dev using the latest.

File: cmd/kured/main.go
Comment (jackfrancis) [reply cmd/kured/main.go:103]: I think the normal pattern is to update the helm chart with a separate PR after the functional change has merged.

Comment (dholbach) [reply cmd/kured/main.go:103]: wfm!

File: README.md
Comment (dholbach) [reply README.md:228]: @atighineanu Can you take a look at this?

Comment (dholbach) [reply README.md:228]: Can this be re-added? Not sure why this section was removed?

Comment (atighineanu) [reply README.md:228]: I will do it today

Comment (atighineanu) [reply README.md:228]: Done.

**Comment by dholbach** [comment]
@atighineanu Thanks for your work on this! The PR looks good to me - can you `git pull --rebase` one more time and fix the conflict?

File: README.md
Comment (hiddeco) [inline README.md:225]: What technologies? Is this list documented somewhere?

Comment (atighineanu) [reply README.md:225]: It's in the same Readme:
```
rocketchat:      `rocketchat://[username@]rocketchat-host/token[/channel|@recipient]`

teams:           `teams://token-a/token-b/token-c`

Email:           `smtp://username:password@host:port/?fromAddress=fromAddress&toAddresses=recipient1[,recipient2,...]`

More details here: https://github.com/containrrr/shoutrrr/blob/master/docs/services/overview.md
```

**Review by hiddeco — APPROVED** [review]

**Pull request merged**
